autogen/oai/openai_utils.py

```diff
@@ -239,3 +243,78 @@ def config_list_from_json(
         except FileNotFoundError:
             return []
     return filter_config(config_list, filter_dict)
+
+
+def config_list_from_dotenv(
+    dotenv_file_path: Optional[str] = None, api_key_env_var: str = "OPENAI_API_KEY", filter_dict: Optional[dict] = None
+) -> List[Dict[str, Union[str, Set[str]]]]:
+    """
+    Loads configuration details from a .env file or from the environment,
+    creates a temporary JSON structure, and sets configurations for autogen.
+
+    Args:
+        dotenv_file_path (str, optional): The path to the .env file. If not provided,
+            it will look for a .env file using the find_dotenv method from the dotenv module.
+        filter_dict (dict, optional): A dictionary containing the models to be loaded.
+            If not provided, it defaults to loading 'gpt-4' and 'gpt-3.5-turbo'.
+        api_key_env_var (str, optional): The name of the environment variable where the API key is stored.
+            Defaults to 'OPENAI_API_KEY'.
+
+    Returns:
+        config_list: A list of configurations loaded from the .env file or the environment.
+        Each configuration is a dictionary containing:
+            - model (str): The model name, e.g., 'gpt-4' or 'gpt-3.5-turbo'.
+            - api_key (str): The API key for OpenAI.
+
+    Raises:
+        ValueError: If no configurations are loaded or if the API key is not found in the environment variables.
+
+    Example:
+        >>> config_list_from_dotenv(dotenv_file_path='path_to_dotenv_file', api_key_env_var='OPENAI_API_KEY')
+        [
+            {'model': 'gpt-4', 'api_key': 'some_api_key'},
+            {'model': 'gpt-3.5-turbo', 'api_key': 'some_api_key'}
+        ]
```

Comment:
Thank you @AaronWard. It looks great to me. What do you think of adding an example of the contents in the `.env` file? I'm asking because I read saw some feedbacks mentioning not knowing what contents should be in the configure files used in the example notebooks.

Reply:
@thinkall Sure, i will make an instructions notebook in [`notebooks directory`](https://github.com/microsoft/autogen/tree/main/notebook) with the different ways of setting up the configurations using:
- `config_list_gpt4_gpt35`
- `config_list_openai_aoai`
- `config_list_from_models`
- `config_list_from_json`

I'll try be as explicit as possible because i think the JSON structure requirement seems to catch people off-guard, particularly for novice developers. I'll get this done later today.

Reply:
Thank you @AaronWard, would you like to add the notebook in this PR?

Also, some tests failed, could you check about it?

Reply:
Working on it now 👍